### PR TITLE
Add sleuth

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [x] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/sleuth/sleuth.xml [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+.. class:: infomark
+
+**Purpose**
+
+Sleuth is a tool for the analysis and comparison of multiple related RNA-Seq experiments. Key features include:
+
+- The ability to perform both transcript-level and gene-level analysis.
+- Compatibility with kallisto enabling a fast and accurate workflow from reads to results.
+- The use of boostraps to ascertain and correct for technical variation in experiments.
+- An interactive app for exploratory data analysis.
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:121]: The interactive app is not part of this tool, isn't it?

File: tools/sleuth/sleuth.xml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+.. class:: infomark
+
+**Purpose**
+
+Sleuth is a tool for the analysis and comparison of multiple related RNA-Seq experiments. Key features include:
+
+- The ability to perform both transcript-level and gene-level analysis.
+- Compatibility with kallisto enabling a fast and accurate workflow from reads to results.
+- The use of boostraps to ascertain and correct for technical variation in experiments.
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:120]: boo__t__strap?

File: tools/sleuth/sleuth.xml [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+- An interactive app for exploratory data analysis.
+
+To use sleuth, RNA-Seq data must first be quantified with kallisto, which is a program for very fast RNA-Seq quantification based on 
+pseudo-alignment. An important feature of kallisto is that it outputs bootstraps along with the estimates of transcript abundances. 
+These can serve as proxies for technical replicates, allowing for an ascertainment of the variability in estimates due to the random 
+processes underlying RNA-Seq as well as the statistical procedure of read assignment. kallisto can quantify 30 million human reads in
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:126]: the entire paragraph with the notebook, Desktop can go imho

File: tools/sleuth/sleuth.xml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+            <param argument="nbins" type="integer" min="0" value="100" label="NBins" help="The number of bins that the data should be 
+                split for the sliding window shrinkage using the mean-variance curve." />
+            <param argument="lwr" type="float" min="0" max="1" value="0.25" label="LWR" help="The lower range of variances within each 
+                bin that should be included for the shrinkage procedure. " />
+            <param argument="upr" type="float" min="0" max="1" value="0.75" label="UPR" help="The upper range of variances within each 
+                bin that should be included for the shrinkage procedure." />
+        </section>
+    </inputs>
+    <outputs>
+        <data name="sleuth_table" from_work_dir="sleuth_table.tab" format="tabular" label="${tool.name} on ${on_string}: DE table"/>
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:85]: add the column names as metadata?

File: tools/sleuth/sleuth.xml [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+                --factorLevel2_counts '${count}'
+            #end for
+            --cores  \${GALAXY_SLOTS:-1}
+            $advanced_options.normalization
+            --nbins $advanced_options.nbins
+            --lwr $advanced_options.lwr
+            --upr $advanced_options.upr
+    ]]></command>
+    <inputs>
+        <section name="first_factor" title="1: Factor level" expanded="true">
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:54]: only two factors are supported? If not maybe a repeat is better?

File: tools/sleuth/sleuth.xml [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+           source="both"
+           level="fatal"
+           description="An undefined error occurred, please check your input carefully and contact your administrator." />
+        <regex match="Fatal error"
+           source="both"
+           level="fatal"
+           description="An undefined error occurred, please check your input carefully and contact your administrator." />
+    </stdio>
+    <version_command><![CDATA[
+echo $(R --version | grep version | grep -v GNU)", DESeq2 version" $(R --vanilla --slave -e "library(DESeq2); cat(sessionInfo()\$otherPkgs\$DESeq2\$Version)" 2> /dev/null | grep -v -i "WARNING: ")
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:23]: DESeq2 ?

File: tools/sleuth/sleuth.xml [hunk 7]
```diff
@@ -0,0 +1,133 @@
+<tool id="sleuth" name="Sleuth" version="@TOOL_VERSION@+galaxy@SUFFIX_VERSION@">
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml:1]: profile?

File: tools/sleuth/sleuth.xml (hunk 1 above)
Comment (gallardoalba) [reply tools/sleuth/sleuth.xml:121]: Yes, the Rscript includes also a interactive app:

```
The easiest way to view and interact with the results is to generate the sleuth live site that allows for exploratory data analysis:

sleuth_live(so)
```

File: tools/sleuth/sleuth.xml (hunk 2 above)
Comment (gallardoalba) [reply tools/sleuth/sleuth.xml:120]: Sorry, I don't undestand this comment. The bootstrap is used by Kallisto, and Sleuth make uses of it. That informaiton is included in the h5 file.

File: tools/sleuth/sleuth.xml (hunk 4 above)
Comment (gallardoalba) [reply tools/sleuth/sleuth.xml:85]: I'm not sure what do you mean sorry.

Comment (bgruening) [reply tools/sleuth/sleuth.xml:85]: See here as an example: https://github.com/galaxyproject/tools-iuc/blob/35db9ac8668f3e376886ea09de63c87dce93e1ce/tools/snipit/snipit.xml#L184

File: tools/sleuth/sleuth.xml (hunk 1 above)
Comment (gallardoalba) [reply tools/sleuth/sleuth.xml:113]: It requires Shine for generating the visualization according my understanding.

Comment (bgruening) [reply tools/sleuth/sleuth.xml:113]: Interesting, we could create a Shiny IT of this app.

File: tools/sleuth/sleuth.xml [hunk 8]
```diff
@@ -19,9 +19,7 @@
            level="fatal"
            description="An undefined error occurred, please check your input carefully and contact your administrator." />
     </stdio>
-    <version_command><![CDATA[
-echo $(R --version | grep version | grep -v GNU)", DESeq2 version" $(R --vanilla --slave -e "library(DESeq2); cat(sessionInfo()\$otherPkgs\$DESeq2\$Version)" 2> /dev/null | grep -v -i "WARNING: ")
-    ]]></version_command>
+    <version_command><![CDATA[ echo $(R --version | grep version | grep -v GNU)"]]></version_command>
```
Comment (bgruening) [inline tools/sleuth/sleuth.xml]: This is now returning only the R version.

File: tools/sleuth/sleuth.xml (hunk 1 above)
Comment (gallardoalba) [reply tools/sleuth/sleuth.xml:113]: In this PR or in a different one?

File: tools/sleuth/sleuth.xml [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+        <regex match="Error in"
+           source="both"
+           level="fatal"
+           description="An undefined error occurred, please check your input carefully and contact your administrator." />
+        <regex match="Fatal error"
+           source="both"
+           level="fatal"
+           description="An undefined error occurred, please check your input carefully and contact your administrator." />
+    </stdio>
+    <version_command><![CDATA[ echo $(R --version | grep version | grep -v GNU)]]></version_command>
```
Comment (gallardoalba) [inline tools/sleuth/sleuth.xml:22]:
```suggestion
    <version_command><![CDATA[echo $(R --version | grep version | grep -v GNU)", sleuth version" $(R --vanilla --slave -e "library(sleuth); cat(sessionInfo()\$otherPkgs\$sleuth\$Version)" 2> /dev/null | grep -v -i "WARNING: ")]></version_command>
```

File: tools/sleuth/sleuth.xml [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        <regex match="Error in"
+           source="both"
+           level="fatal"
+           description="An undefined error occurred, please check your input carefully and contact your administrator." />
+        <regex match="Fatal error"
+           source="both"
+           level="fatal"
+           description="An undefined error occurred, please check your input carefully and contact your administrator." />
+    </stdio>
+    <version_command><![CDATA[echo $(R --version | grep version | grep -v GNU)", sleuth version" $(R --vanilla --slave -e "library(sleuth); cat(sessionInfo()\$otherPkgs\$sleuth\$Version)" 2> /dev/null | grep -v -i "WARNING: ")]></version_command>
```
Comment (gallardoalba) [inline tools/sleuth/sleuth.xml:22]:
```suggestion
    <version_command><![CDATA[echo $(R --version | grep version | grep -v GNU)", sleuth version" $(R --vanilla --slave -e "library(sleuth); cat(sessionInfo()\$otherPkgs\$sleuth\$Version)" 2> /dev/null | grep -v -i "WARNING: ")]]></version_command>
```

**Review by bgruening — APPROVED** [review]

**Pull request merged**